astropy/modeling/core.py

```diff
@@ -2703,31 +2747,54 @@ def _model_evaluate_getter(idx, model):
         n_inputs = model.n_inputs
         n_outputs = model.n_outputs
 
-        # There is currently an unfortunate inconsistency in some models, which
-        # requires them to be instantiated for their evaluate to work.  I think
-        # that needs to be reconsidered and fixed somehow, but in the meantime
-        # we need to check for that case
-        if (not isinstance(model, Model) and
-                isinstancemethod(model, model.evaluate)):
+        def process_output_units(model, inputs, outputs):
```

Comment:
This duplicates some code with the block at the end of ``evaluate`` - I think we should instead make a separate method to handle this like we did for input units and call that below.